include/evmmax/evmmax.hpp

```diff
@@ -0,0 +1,42 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include <intx/intx.hpp>
+
+namespace evmmax
+{
+
+/// The modular arithmetic operations for EVMMAX (EVM Modular Arithmetic Extensions).
+template <typename UintT>
+class ModArith
+{
+public:
+    using uint = UintT;
+
+    const uint mod;  ///< The modulus.
+
+private:
+    const uint m_r_squared;    ///< R² % mod.
+    const uint64_t m_mod_inv;  ///< The modulus inversion.
+
+public:
+    explicit ModArith(const UintT& modulus) noexcept;
```

Comment:
Why `UintT` here but the new type in others?

Reply:
No reason. The methods were added at different times.

Reply:
Fixed.